api/options.c

```diff
@@ -270,7 +270,7 @@ nlopt_result
 NLOPT_STDCALL nlopt_set_lower_bounds(nlopt_opt opt, const double *lb)
 {
      nlopt_unset_errmsg(opt);
-     if (opt && (opt->n == 0 || lb)) {
+     if (opt && (opt->n > 0) && lb) {
```

Comment:
oh, then only the set methods are affecteor, or this can be done with just an "if"